src/Symfony/Component/Notifier/Bridge/Bluesky/BlueskyTransport.php

```diff
@@ -0,0 +1,227 @@
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Component\String\AbstractString;
+use Symfony\Component\String\ByteString;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\DecodingExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+final class BlueskyTransport extends AbstractTransport
+{
+    private array $authSession = [];
+
+    public function __construct(
+        #[\SensitiveParameter]
+        private string $user,
+        #[\SensitiveParameter]
+        private string $password,
+        private LoggerInterface $logger,
+        HttpClientInterface $client = null,
+        EventDispatcherInterface $dispatcher = null,
+    ) {
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('bluesky://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+
+        if ([] === $this->authSession) {
+            $this->authenticate();
+        }
+
+        $post = [
+            '$type' => 'app.bsky.feed.post',
+            'text' => $message->getSubject(),
+            'createdAt' => (new \DateTimeImmutable())->format('Y-m-d\\TH:i:s.u\\Z'),
+        ];
+        if ([] !== $facets = $this->parseFacets($post['text'])) {
+            $post['facets'] = $facets;
+        }
+
+        $response = $this->client->request('POST', sprintf('https://%s/xrpc/com.atproto.repo.createRecord', $this->getEndpoint()), [
+            'auth_bearer' => $this->authSession['accessJwt'] ?? null,
+            'json' => [
+                'repo' => $this->authSession['did'] ?? null,
+                'collection' => 'app.bsky.feed.post',
+                'record' => $post,
+            ],
+        ]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the remote bluesky server.', $response, 0, $e);
+        }
+
+        if (200 === $statusCode) {
+            $content = $response->toArray();
+            $sentMessage = new SentMessage($message, (string) $this);
+            $sentMessage->setMessageId($content['cid']);
+
+            return $sentMessage;
+        }
+
+        try {
+            $content = $response->toArray(false);
```

Comment:
`false` means it will not throw exceptions, isn't it?

Reply:
Yes, correct. 

I check if status code is 200 just before. So this try-catch is only for decoding error